File: app/lib/domain/models/user_profile.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_profile.freezed.dart';
part 'user_profile.g.dart';

enum Goal { muscleGain, fatLoss, maintenance, strength }
enum Experience { beginner, intermediate, advanced }
enum Sex { male, female, other }

@freezed
abstract class UserProfile with _$UserProfile {
  const factory UserProfile({
    required int age,
    @JsonKey(name: 'height_cm') required double heightCm,
    @JsonKey(name: 'weight_kg') required double weightKg,
    required Sex sex,
    required Goal goal,
    required Experience experience,
    @Default([]) List<String> equipment,
    String? injuries,
    @JsonKey(name: 'dietary_notes') String? dietaryNotes,
    @JsonKey(name: 'daily_schedule') @Default({}) Map<String, bool> dailySchedule,
    @JsonKey(name: 'updated_at') required DateTime updatedAt,
  }) = _UserProfile;

  factory UserProfile.fromJson(Map<String, dynamic> json) =>
      _$UserProfileFromJson(json);
}

File: app/android/app/src/main/java/com/gymcoach/gym_coach/domain/model/UserProfile.kt
package com.gymcoach.gym_coach.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Mirrors `lib/domain/models/user_profile.dart`. Enums are serialized via
 * `name.toCamelCase()` style keys (`muscle_gain`, `fat_loss`, etc.) to match
 * what the LLM emits; the Flutter code used `Sex.values.byName` which
 * matched on the JSON string. Same contract here.
 */
@Serializable
data class UserProfile(
    val age: Int,
    @SerialName("height_cm") val heightCm: Double,
    @SerialName("weight_kg") val weightKg: Double,
    val sex: Sex,
    val goal: Goal,
    val experience: Experience,
    val equipment: List<String> = emptyList(),
    val injuries: String? = null,
    @SerialName("dietary_notes") val dietaryNotes: String? = null,
    @SerialName("daily_schedule") val dailySchedule: Map<String, Boolean> = emptyMap(),
    @SerialName("updated_at") val updatedAt: Long = System.currentTimeMillis()
)
